ingest/ingestHelpers.py

```diff
@@ -139,121 +125,46 @@ def get_distributions(ds):
-    # parser.add_argument('--es', default="https://dcotest.tw.rpi.edu/search", help="elasticsearch service URL")
     parser.add_argument('--publish', default=False, action="store_true", help="publish to elasticsearch?")
     parser.add_argument('--rebuild', default=False, action="store_true", help="rebuild elasticsearch index?")
-    parser.add_argument('--mapping', default="mappings/dataset.json", help="dataset elasticsearch mapping document")
+    parser.add_argument('--mapping', help="elasticsearch mapping document, e.g. mappings/dataset.json")
     parser.add_argument('--sparql', default='http://deepcarbon.tw.rpi.edu:3030/VIVO/query', help='sparql endpoint')
-    # parser.add_argument('--sparql', default='http://udco.tw.rpi.edu/fuseki/vivo/query', help='sparql endpoint')
     parser.add_argument('out', metavar='OUT', help='elasticsearch bulk ingest file')
 
+    # Info:
+    #   local elasticsearch URL:          http://localhost:9200
+    #   dcotest elasticsearch URL:        https://dcotest.tw.rpi.edu/search
+    #   production sparql endpoint:       http://deepcarbon.tw.rpi.edu:3030/VIVO/query
+    #   test sparql endpoint:             http://udco.tw.rpi.edu/fuseki/vivo/query
+
     args = parser.parse_args()
 
-    # generate bulk import document for datasets
-    records = generate(threads=int(args.threads), sparql=args.sparql, get_objects_query=get_objects_query,
-                       process_object_function=process_dataset, create_object_doc_function=create_object_doc_function,
-                       object_index=object_index, object_type=object_type)
+    ingestSomething = XIngest(elasticsearch_index=elasticsearch_index, elasticsearch_type=elasticsearch_type,
+                             get_objects_query_location=get_objects_query_location,
+                             describe_object_query_location=describe_object_query_location,
+                             variable_name_sparql=variable_name_sparql)
+
+    # if a mapping file is specified for the "publish" process later, use the specified mapping file
+    if args.mapping:
+        ingestSomething.setMapping(args.mapping)
+
+    # generate bulk import document for xs
+    ingestSomething.generate(threads=int(args.threads), sparql=args.sparql)
 
     # save generated bulk import file so it can be backed up or reviewed if there are publish errors
     with open(args.out, "w") as bulk_file:
-        bulk_file.write('\n'.join(records))
+        bulk_file.write('\n'.join(ingestSomething.records))
 
     # publish the results to elasticsearch if "--publish" was specified on the command line
     if args.publish:
-        bulk_str = '\n'.join(records)
-        publish(bulk=bulk_str, endpoint=args.es, rebuild=args.rebuild, mapping=args.mapping, index=object_index, tYPE=object_type)
-    print(get_objects_query)
\ No newline at end of file
+        bulk_str = '\n'.join(ingestSomething.records)
```

Comment:
In a recent branch we added a newline to bulk_str to make sure all documents are added to elasticsearch.

bulk_str = '\n'.join(records)+'\n'

Please make that change